sql/src/main/java/io/crate/metadata/Routing.java

```diff
@@ -128,6 +128,42 @@ public boolean containsShards(String nodeId) {
         return false;
     }
 
+    public void updateLocations(Map<String, Map<String, List<Integer>>> existingLocations) {
```

Comment:
i think you should add a test with a self join where the routing column is used with different queries